### PR TITLE
appium(docs): Add dotnet quickstart documentation

### PR DESCRIPTION
**Pull request by mahomedalid**
## Proposed changes

This Pull Request (PR) seeks to enhance the Appium documentation by adding a comprehensive Quickstart guide for the .NET community. The .NET community is known for its vibrancy, filled with passionate and skilled engineers, and we do not want to alienate them.

Why This PR Matters:

* Vibrant .NET Community: The .NET community is vast, diverse, and ever-growing. It's home to numerous developers, testers, and engineers who have a strong passion for creating high-quality applications. 

* Welcoming Smart Engineers: .NET developers are known for their expertise in building robust and scalable applications. By accommodating the .NET community, Appium invites some of the smartest engineers to contribute, collaborate, and improve the framework. Their expertise can lead to advancements and enhancements that benefit all Appium users.

* Increased Adoption: Expanding the reach of Appium to the .NET community can lead to an increase in its adoption. As a result, more organizations and individuals will use Appium for mobile app testing, thus strengthening the Appium ecosystem. The dotnet-client repo needs some love as well.

* Mutually Beneficial: This PR is a win-win situation. Appium gains talented contributors and a larger user base, while the .NET community benefits from an easier entry point to Appium for their mobile app testing needs.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mahomedalid** [comment]
cc @Dor-bl

File: packages/appium/docs/en/quickstart/test-dotnet.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+---
+
+The [Appium dotnet Client](https://github.com/appium/dotnet-client/) is
+an official Appium client in C#. This driver is an extension of the Selenium C# client. It has all the functionalities of the regular driver, but add Appium-specific methods on top of this. The driver is not available under Nuget, since some of the dependencies do not have a signed version. With that said, you are more than welcome to download the code and build a signed version yourself. For the purposes of this quickstart we will proceed to clone the project code.
+
+```bash
+git clone https://github.com/appium/dotnet-client.git
+```
+
+Now, we get inside the directory and create a new [NUnit](https://nunit.org/) project. Ww will also add the references to the Appium.Net driver, and other dependencies.
```
Comment (Dor-bl) [inline packages/appium/docs/en/quickstart/test-dotnet.md:12]: A small typo on Ww

File: packages/appium/docs/en/quickstart/test-dotnet.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+title: Write a Test (dotnet)
+---
+
+The [Appium dotnet Client](https://github.com/appium/dotnet-client/) is
+an official Appium client in C#. This driver is an extension of the Selenium C# client. It has all the functionalities of the regular driver, but add Appium-specific methods on top of this. The driver is not available under Nuget, since some of the dependencies do not have a signed version. With that said, you are more than welcome to download the code and build a signed version yourself. For the purposes of this quickstart we will proceed to clone the project code.
```
Comment (Dor-bl) [inline packages/appium/docs/en/quickstart/test-dotnet.md:6]: What do you mean by the driver is not available under NuGet?

File: packages/appium/docs/en/quickstart/test-dotnet.md (hunk 1 above)
Comment (mahomedalid) [reply packages/appium/docs/en/quickstart/test-dotnet.md:12]: Thanks, fix it.

File: packages/appium/docs/en/quickstart/test-dotnet.md (hunk 2 above)
Comment (mahomedalid) [reply packages/appium/docs/en/quickstart/test-dotnet.md:6]: I literally copy-pasted this disclaimer from https://github.com/appium/dotnet-client/

image

A quick search in Nuget got me this package (https://www.nuget.org/packages/Appium.WebDriver/5.0.0-rc.1) which seems like the official one and I do see you as contributor. I will give it a try and update this PR. Should we also update the dotnet-client one?

Comment (Dor-bl) [reply packages/appium/docs/en/quickstart/test-dotnet.md:6]: Not sure about this disclaimer, maybe @laolubenson recall what it's about.
Anyway, the version you sent it's the official release of the project. I still need to release RC2 in the future, maybe even this week.

Comment (mahomedalid) [reply packages/appium/docs/en/quickstart/test-dotnet.md]: @Dor-bl  fixing doc, adding link to the nuget version. Confirming it continue to works.

image

Let me know if you need help with rc2, coding or testing.

Comment (Dor-bl) [reply packages/appium/docs/en/quickstart/test-dotnet.md]: Cool, thanks for the effort.

File: packages/appium/docs/en/quickstart/test-dotnet.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        driverOptions.AddAdditionalAppiumOption("noReset", true);
+
+        _driver = new AndroidDriver(serverUri, driverOptions, TimeSpan.FromSeconds(180));
+        _driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);
+    }
+
+    [SetUp]
+    public void Setup()
+    {
+    }
```
[the remark this answers is not in the document]
Comment (mahomedalid) [reply packages/appium/docs/en/quickstart/test-dotnet.md:57]: Sorry, I miss this, it is not, I think it was just leftovers of the original generated UnitTest.cs. Updated the code and adding TearDown.

**Review by Dor-bl — APPROVED** [review]
LGTM after the changes.
Nice work

**Comment by Dor-bl** [comment]
@mykola-mokhnach /@jlipps, anything you want to add?

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
fine with me. Thanks for checking it @Dor-bl

**Pull request merged**